osu.Game/Screens/Play/SquareGraph.cs

```diff
@@ -78,18 +78,21 @@ public Color4 FillColour
         private readonly LayoutValue layout = new LayoutValue(Invalidation.DrawSize);
         private ScheduledDelegate scheduledCreate;
 
+        private Vector2 previousDrawSize = new Vector2();
+
         protected override void Update()
         {
             base.Update();
 
-            if (values != null && !layout.IsValid)
+            if (values != null && !layout.IsValid && DrawSize != previousDrawSize)
```

Comment:
`!layout.IsValid && DrawSize != previousDrawSize` is equivalent to "if and only if `this.DrawSize` Changed", this does not look very clean but it is the only way i found to check for this condition.
an alternative to this solution would be to wrap SquareGraph/SongProgressGraph with another container so that `layout` doesn't get invalidated when the size of `DefaultSongProgress`(its current parent) changes.